source/auth/tests/mongodb-oidc.rst

```diff
@@ -4,164 +4,386 @@ MongoDB OIDC
+
+Drivers will need to be able to internally query and clear the cached
+credentials to verify usage for testing purposes.  Clearing the cache
+means removing all data from the cache, including ``OIDCMechanismServerStep1``
+information.
+
+Drivers MUST set the ``AWS_WEB_IDENTITY_TOKEN_FILE`` environment variable
+to the location of valid ``test_user1`` credentials.
+
+Unless otherwise specified, tests will use a URL
+of the form ``mongodb://localhost/?authMechanism=MONGODB-OIDC``.
+
 
 Callback-Driven Auth
 ====================
 
 Drivers MUST be able to authenticate using OIDC callback(s) when there
 is one principal configured.
 
-#. Create a request callback that reads in the generated ``test_user1`` token
-file.
-#. Create a client with a url of the form  ``mongodb://localhost/?authMechanism=MONGODB-OIDC`` and the OIDC request callback.
-#. Perform a ``find`` operation.
-#. Clear the cache.
+Single Principal Implicit Username
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a request callback returns a valid token.
+- Create a client that uses the default OIDC url and the request callback.
+- Perform a ``find`` operation. that succeeds.
+- Close the client.
+
+Single Principal Explicit Username
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a request callback that returns a valid token.
+- Create a client with a url of the form  ``mongodb://test_user1@localhost/?authMechanism=MONGODB-OIDC`` and the OIDC request callback.
+- Perform a ``find`` operation that succeeds.
+- Close the client.
+
+Multiple Principal User 1
+~~~~~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a request callback that returns a valid token.
+- Create a client with a url of the form  ``mongodb://test_user1@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Perform a ``find`` operation that succeeds.
+- Close the client.
+
+Multiple Principal User 2
+~~~~~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a request callback that reads in the generated ``test_user2`` token file.
+- Create a client with a url of the form ``mongodb://test_user2@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Perform a ``find`` operation that succeeds.
+- Close the client.
+
+Multiple Principal No User
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a client with a url of the form ``mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Assert that a ``find`` operation fails.
+- Close the client.
+
+OIDC Allowed Hosts Blocked
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a client that uses the OIDC url and the request callback, and an
+  ``authOIDCAllowHosts`` that is an empty list.
```

Comment:
also, now it's called `ALLOWED_HOSTS` auth property

Reply:
Fixed